src/main/java/in/twizmwaz/cardinal/module/modules/bossBar/BossBar.java

```diff
@@ -75,45 +88,127 @@ FakeWither getWither(Player player, String message) {
         if (players.containsKey(player)) {
             return players.get(player);
         } else {
-            return addWither(player, checkMessageLength(message));
+            return addWither(player, checkMessageLength(message), false);
         }
     }
 
-    FakeWither addWither(Player player, String message) {
+    FakeWither addWither(Player player, Location location, String message) {
         FakeWither wither = new FakeWither(message, getWitherLocation(player));
-        ((CraftPlayer) player).getHandle().playerConnection.sendPacket(wither.getSpawnPacket());
+        if (wither.isVisible()) {
+            ((CraftPlayer) player).getHandle().playerConnection.sendPacket(wither.getSpawnPacket());
+        }
         players.put(player, wither);
         return wither;
     }
 
-    private FakeWither addWither(Player player, Location loc, String message) {
+    private FakeWither addWither(Player player, String message, boolean visible) {
         FakeWither wither = new FakeWither(message, getWitherLocation(player));
-        ((CraftPlayer) player).getHandle().playerConnection.sendPacket(wither.getSpawnPacket());
+        wither.setVisible(visible);
+        if (visible) {
+            ((CraftPlayer) player).getHandle().playerConnection.sendPacket(wither.getSpawnPacket());
+        }
         players.put(player, wither);
         return wither;
     }
 
     private Location getWitherLocation(Player player) {
-        return player.getLocation().add(player.getEyeLocation().getDirection().multiply(100));
+        if (GameHandler.getGameHandler().getMatch().isRunning()) {
```

Comment:
@Alfalik What is the point of this if/else statement? You're returning the same thing for both cases, if this is the case you don't even need the statement.